addons/sale_purchase_stock/models/purchase.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    mto_sale_order_count = fields.Integer('Number of source sale orders ', compute='_compute_mto_sale_order_count', store=True,
+                                          groups='sales_team.group_sale_salesman')
+
+    @api.depends('order_line.move_dest_ids.group_id.sale_id')
+    def _compute_mto_sale_order_count(self):
+        for purchase in self:
+            purchase.mto_sale_order_count = len(purchase.order_line.move_dest_ids.group_id.sale_id)
+
+    def action_view_mto_sale_orders(self):
```

Comment:
could use something like delivery and directly display the form view if there is only 1 SO